src/modules/hunting/apiserver.py

```diff
@@ -0,0 +1,716 @@
+    specific role's namespace scope
+    """
+
+    def __init__(self, evidence):
+        Vulnerability.__init__(self, KubernetesCluster, name="Patched a role",
+                               category=AccessRisk)
+        self.evidence = evidence
+
+
+class PatchAClusterRole(Vulnerability, Event):
+    """ Patching a cluster role might give an attacker the option to create new pods with custom roles within the whole
+    cluster scope.
+    """
+
+    def __init__(self, evidence):
+        Vulnerability.__init__(self, KubernetesCluster, name="Patched a cluster role",
+                               category=AccessRisk)
+        self.evidence = evidence
+
+
+class DeleteARole(Vulnerability, Event):
+    """ Deleting a role might allow an attacker to affect access to resources in the namespace"""
+
+    def __init__(self, evidence):
+        Vulnerability.__init__(self, KubernetesCluster, name="Deleted a role",
+                               category=AccessRisk)
+        self.evidence = evidence
+
+
+class DeleteAClusterRole(Vulnerability, Event):
+    """ Deleting a cluster role might allow an attacker to affect access to resources in the cluster"""
+
+    def __init__(self, evidence):
+        Vulnerability.__init__(self, KubernetesCluster, name="Deleted a cluster role",
+                               category=AccessRisk)
+        self.evidence = evidence
+
+
+class CreateAPod(Vulnerability, Event):
+    """ Creating a new pod allows an attacker to run custom code"""
+
+    def __init__(self, evidence):
+        Vulnerability.__init__(self, KubernetesCluster, name="Created A Pod",
+                               category=AccessRisk)
+        self.evidence = evidence
+
+
+class PatchAPod(Vulnerability, Event):
+    """ Patching a pod allows an attacker to compromise and control it """
+
+    def __init__(self, evidence):
+        Vulnerability.__init__(self, KubernetesCluster, name="Patched A Pod",
+                               category=AccessRisk)
+        self.evidence = evidence
+
+
+class DeleteAPod(Vulnerability, Event):
+    """ Deleting a pod allows an attacker to disturb applications on the cluster """
+
+    def __init__(self, evidence):
+        Vulnerability.__init__(self, KubernetesCluster, name="Deleted A Pod",
+                               category=AccessRisk)
+        self.evidence = evidence
+
+
+class ApiServerPassiveHunterFinished(Event):
+    def __init__(self, all_namespaces_names, service_account_token, host, port):
+        self.host = host
+        self.port = port
+        self.all_namespaces_names = all_namespaces_names
+        self.service_account_token = service_account_token
+
+    def __str__(self):
+        return str(self.service_account_token)
+
+
+# Passive Hunter
+@handler.subscribe(OpenPortEvent, predicate=lambda x: x.port == 443 or x.port == 6443)
+class AccessApiServerViaServiceAccountToken(Hunter):
```

Comment:
I think I'd have been inclined to split this into 
* a hunter that checks whether you can read the service account token and access the API Server with it (which publishes a ServerApiAccess event) 
* another hunter (or maybe more than one) that subscribe to the ServerApiAccess event 

Just because then each hunter would be a bit easier to read independently